internal/scorer/algorithm/wam/wam.go

```diff
@@ -18,16 +18,26 @@
 package wam
 
 import (
+	"errors"
+
 	"github.com/ossf/criticality_score/internal/scorer/algorithm"
 )
 
+var ErrWeightLessThanOrEqualToZero = errors.New("weight must be greater than 0")
+
 type WeighetedArithmeticMean struct {
 	inputs []*algorithm.Input
 }
 
 // New returns a new instance of the Weighted Arithmetic Mean algorithm, which
 // is used by the Pike algorithm.
 func New(inputs []*algorithm.Input) (algorithm.Algorithm, error) {
+	for _, i := range inputs {
+		if i.Weight <= 0 {
+			return nil, ErrWeightLessThanOrEqualToZero
+		}
+	}
+
```

Comment:
This check belongs better in `config.go` or a new function for constructing Input instances.

Reply:
I don't understand what you mean, could you please abbreviate?

Reply:
Sure.

Currently, this check will only be applied to the Weighted Arithmetic Mean.

The check should be applied across all possible algorithms as zero weights are no-ops and negative weights are confusing and error prone.

The most sensible place to have this check is inside the config.go in `UnmarshalYAML()` while the Input is being parsed.

However, another place it could be added is in `algorithm/input.go` by creating a 'NewInput()` method that does the validation which would be called from `ToAlgorithmInput()` in config.go.

Reply:
Thank you for the explanation. I have added the check into `UnmarshalYAML()`.